[Docs] Add example of build_image using project source (#4916)
mlrun/mlrun: docs/projects/run-build-deploy.md

The {py:meth}`~mlrun.projects.MlrunProject.build_image` function builds an image using the existing build configuration. 
This method can also be used to set the build configuration and build the image based on it - in a single step. 

If you set a source for the project (for example, git source) and set `pull_at_runtime = False`, then 

> the images used contain the project source in them. For example, this code builds `.some-project-image` 

Reviewer: ```suggestion
the generated image contains the project source in it. For example, this code builds `.some-project-image` 
```

image with the source in it.
```python
project = mlrun.get_or_create_project(
    name="project-name", context="./"
)

project.set_source(
    "git://some/repo",
    pull_at_runtime=False
)

project.build_image(image=".some-project-image")
```

> And now you can run a function without having to specify an image:

Reviewer: ```suggestion
And now you can run a function based on the project code without having to specify an image:
```
